### PR TITLE
Implement SDKv4 multisignature api - Closes #142

### LINKED ISSUE #142: Implement SDKv4 multisignature API
### Description

Make the blockchain compatibility layer able to retrieve SDKv4 multisignature account information. Keep compatibility with older versions of Lisk Core/SDK.

### Motivation

- The UI products require working multisignature accounts

### Acceptance Criteria

- Multisignature account properties are available
- Previous multisignature accounts work without regressions
- Add support for multisig balance transfers

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #142 

### How was it solved?
- [x] Multisignature account properties are available for SDKv4
- [x] Previous multisignature accounts work without regressions
- [x] Add support for multisig balance transfers

### How was it tested?
Locally

### CODE REVIEW

File: tests/integration_testnet/api/http/delegates.test.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ const delegateSchema = {
 	secondPublicKey: 'string',
 	username: 'string',
 	vote: 'string',
-	status: 'string',
+	// status: 'string', uncomment when testnet updated to SDKv4
```
Comment (sameersubudhi) [inline tests/integration_testnet/api/http/delegates.test.js:38]: `status` is now a mandatory property. Kindly enable it in the schema.

```suggestion
	status: 'string',
```

File: services/gateway/sources/mappings/account.js [hunk 2]
```diff
@@ -55,4 +67,9 @@ module.exports = {
 		incoming: 'incomingTxsCount,string',
 		outgoing: 'outgoingTxsCount,string',
 	},
+	unlocking: ['unlocking', {
+		amount: '=,string',
+		height: 'unvoteHeight,number',
```
Comment (sameersubudhi) [inline services/gateway/sources/mappings/account.js:72]: As per the `SDKv4 Impact Analysis on Lisk Service` document, `height` must be an object with the properties `start` and `end`. Kindly make necessary changes to reflect the same.

Perhaps it'll result in something similar to as suggested below. Kindly verify and modify as necessary:
```suggestion
		height: {
			start: 'unvoteHeight.start,string',
			end: 'unvoteHeight.end,string',
		}
```

File: services/gateway/sources/mappings/account.js [hunk 3]
```diff
@@ -35,12 +45,14 @@ module.exports = {
...
 		lifetime: 'multisignatureGroups.lifetime,number', // returns no data at a delegate account
 		minimalNumberAcccounts: 'multisignatureGroups.min,number', // returns no data at a delegate account
+		numberOfReqSignatures: 'multisignatureGroups.numberOfReqSignatures,number',
 		members: ['multisignatureGroups.members', {
 			address: '=,string',
 			publicKey: '=,string',
 			secondPublicKey: '=,string',
 			balance: '=,number',
 			unconfirmedSignature: '=,number',
+			isMandatory: '=',
```
Comment (sameersubudhi) [inline services/gateway/sources/mappings/account.js:55]:
```suggestion
			isMandatory: '=,boolean',
```

File: services/gateway/sources/mappings/account.js [hunk 4]
```diff
@@ -27,6 +28,15 @@ module.exports = {
 		rewards: '=,string',
 		username: '=,string',
 		vote: '=,string',
+		totalVotesReceived: ',string',
+		isBanned: '=',
```
Comment (sameersubudhi) [inline services/gateway/sources/mappings/account.js:32]:
```suggestion
		isBanned: '=,boolean',
```

File: services/gateway/sources/mappings/account.js [hunk 5]
```diff
@@ -27,6 +28,15 @@ module.exports = {
 		rewards: '=,string',
 		username: '=,string',
 		vote: '=,string',
+		totalVotesReceived: ',string',
```
Comment (sameersubudhi) [inline services/gateway/sources/mappings/account.js:31]: `SDKv4 Impact Analysis on Lisk Service` document does not reflect this to be a required change. Request @MichalTuleja to confirm.

File: services/core/shared/core/accounts.js [hunk 6]
```diff
@@ -57,7 +60,20 @@ const getTopAccounts = () => new Promise((resolve) => {
 		resolve(topAccounts);
 	});
 
-const getAccounts = async (params) => {
+const resolveDelegateInfoByAddress = async accounts => {
+	await BluebirdPromise.map(
+		accounts.data, async account => {
+			if (account.isDelegate) {
+				// getDelegates takes care of retreive delegates from cache first
```
Comment (sameersubudhi) [inline services/core/shared/core/accounts.js:67]:
```suggestion
```

File: services/core/shared/core/compat/sdk_v2/accounts.js [hunk 7]
```diff
@@ -73,5 +79,25 @@ const getAccounts = async params => {
 	return result;
 };
 
+const getMultisignatureGroups = async account => {
+	const result = await coreApi.getMultisignatureGroups(parseAddress(account.address));
+	return isProperObject(result) && Array.isArray(result.data)
+		? result.data[0]
+		: [];
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v2/accounts.js:86]: Recommend to make it a single liner, if the linter permits:

```suggestion
	return isProperObject(result) && Array.isArray(result.data) ? result.data[0] : [];
```

else try:

```suggestion
	return isProperObject(result) && Array.isArray(result.data)
		? result.data[0] : [];
```

File: services/core/shared/core/compat/sdk_v2/accounts.js [hunk 8]
```diff
@@ -73,5 +79,25 @@ const getAccounts = async params => {
...
+};
+
+const getMultisignatureMemberships = async (account) => {
+	const result = await coreApi.getMultisignatureMemberships(
+		parseAddress(account.address),
+	);
+	return isProperObject(result) && Array.isArray(result.data)
+		? result.data
+		: [];
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v2/accounts.js:95]: Same as above!

File: tests/integration_testnet/api/http/delegates.test.js (hunk 1 above)
Comment (nagdahimanshu) [reply tests/integration_testnet/api/http/delegates.test.js]: done!

File: services/gateway/sources/mappings/account.js (hunk 3 above)
Comment (nagdahimanshu) [reply services/gateway/sources/mappings/account.js]: done!

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please consider the recommended changes.

File: services/gateway/sources/mappings/account.js (hunk 5 above)
Comment (nagdahimanshu) [reply services/gateway/sources/mappings/account.js]: @MichalTuleja what do you recommend?

File: services/core/shared/core/accounts.js (hunk 6 above)
Comment (nagdahimanshu) [reply services/core/shared/core/accounts.js]: Removed.

File: services/gateway/sources/mappings/account.js (hunk 5 above)
Comment (nagdahimanshu) [reply services/gateway/sources/mappings/account.js]: Given property removed for now based on `SDKv4 Impact Analysis on Lisk Service`

File: services/core/shared/core/compat/sdk_v2/accounts.js (hunk 7 above)
Comment (nagdahimanshu) [reply services/core/shared/core/compat/sdk_v2/accounts.js]: Done!

File: services/gateway/sources/mappings/account.js (hunk 2 above)
Comment (nagdahimanshu) [reply services/gateway/sources/mappings/account.js]: @MichalTuleja @sameersubudhi : Got clarification on this from @UsamaHameed. Structure `UI` need from lisk-service for unlocking is:
```unlocking: [{ 
        amount: '1000000000',
        unvoteHeight: 280000, 
        delegateAddress: '1L' 
}]```
In frontend, they already take care of this.

**Review by MichalTuleja — APPROVED** [review]

**Review by sameersubudhi — APPROVED** [review]
Looks good 👍

**Pull request merged**